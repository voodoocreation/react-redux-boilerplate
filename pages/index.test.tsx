import { render } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";

import Index from "./index";

jest.mock("../src/helpers/withRedux");

const setup = (fn: any) => {
  const props = {
    apiData: {
      serverTest: "Test"
    },
    localData: {
      inputValue: "Test"
    }
  };

  return {
    actual: fn(<Index {...props} />),
    props
  };
};

describe("<Index />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);
    expect(toJson(actual)).toMatchSnapshot();
  });
});
