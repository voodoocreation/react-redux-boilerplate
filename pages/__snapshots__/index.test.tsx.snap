// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> renders correctly 1`] = `
<article>
  <h1>
    <span>
      Welcome to the React Redux Boilerplate
    </span>
  </h1>
  <p>
    <span>
      Some basic examples of user interaction are shown below.
    </span>
  </p>
  <div
    class="row"
  >
    <section
      class="col-sm-6"
    >
      <h2>
        <span>
          API data
        </span>
      </h2>
      <pre>
        {
  "serverTest": "Test"
}
      </pre>
      <button
        class="btn btn-primary"
      >
        <span>
          Fetch API data
        </span>
      </button>
    </section>
    <section
      class="col-sm-6"
    >
      <h2>
        <span>
          Local data
        </span>
      </h2>
      <pre>
        {
  "inputValue": "Test"
}
      </pre>
      <div
        class="input-group"
      >
        <input
          class="form-control"
          type="text"
          value="Test"
        />
        <span
          class="input-group-btn"
        >
          <button
            class="btn btn-primary"
          >
            <span>
              Update local data
            </span>
          </button>
        </span>
      </div>
    </section>
  </div>
</article>
`;
